### ISSUE TITLE
critical error using renaming in traits

### ISSUE THREAD

**Issue by ogmueller** [pdepend/pdepend#320]
When using the following code in PHP, which is syntactically correct, pdepend v2.5.0 will show a critical error:

```PHP
trait Timestampable
{
    use TimestampableCreated {
        initTimestampable as initCreateTimestampable;
    }
    use TimestampableModified {
        initTimestampable as initModifiedTimestampable;
    }

    /**
     * Initialize timestamps
     */
    protected function initTimestampable()
    {
        $this->initCreateTimestampable();
    }

}
```

This is the error output from pdepend:
```
Critical error: 
=============== 
Trait method initTimestampable has not been applied, because there are collisions with other trait methods on Demo\Timestampable.
```

**Comment by kylekatarnls** [comment]
Now that I fixed #458, I will check if this issue may be related.

**Comment by kylekatarnls** [comment]
As not specified, I tried last the version published before your issue post (so PHPMD 2.5.0 / PDepend 2.4.0) and I re-tested with last stable versions, none gave me any error with the following code:

```php
<?php

trait A {
    function test() {
        echo 'a';
    }
}

trait B {
    function test() {
        echo 'b';
    }
}

class C {
    use A {
        A::test as testA;
    }
    use B {
        B::test as testB;
    }

    function test() {
        $this->testA();
        $this->testB();
    }
}

$c = new C;

$c->test();
```

So I suspect there is no trouble in your `Timestampable` trait but you more likely get the issue #367 from inside `TimestampableCreated` or `TimestampableModified`.

So please try to require `"pdepend/pdepend": "dev-feature/issue-367-fix-trait-instead"` and check if it solves your issue.

Thanks and sorry for the delay, the new maintainers teams is catching up the years issue by issue.
